Guide to B2B product metrics (#3583)
PostHog/posthog.com: contents/blog/b2b-saas-product-metrics.md

Read our [product page](/product) and [documentation](/docs) if you're interested.

> This article is part of our [PostHog Academy series](/blog/categories/posthog-academy) where we explain the fundamentals of product analytics.

## What makes a good B2B product metric?

Good metrics are understandable, comparative, specific and actionable.

Here's what those mean in a little more detail:

1. **Understandable:** Can you easily explain a metric to anyone in the business? If you can't, it's unlikely people will want to use or refer to it.

2. **Comparative:** Can you benchmark a metric against your industry, or compare internally using cohorts?

3. **Specific:** Some broad trends are useful, but we recommend metrics that measure specific actions. Don't measure how long users use your product, measure how long it takes for them to get value from it.

4. **Actionable:** Can you easily identify actions to take when you see a drop in a metric? If yes, you're onto a winner. Specific metrics are the most actionable metrics, too.

> It's not realistic achieve all four in every metric you use, but three out of four is a good benchmark.

Reviewer: ```suggestion
It's not realistic to achieve all four in every metric you use, but three out of four is a good benchmark.
```

**Is it useful?** It's important to track, but it's best used in combination with other metrics, or to calculate other more meaningful metrics.

### Adoption rate

**What is it?**  Adoption rate is a product health metric that tells you what percentage of total signups turned into active users in any given period. 

Your adoption rate is `(new active users / signups) x 100`, so 40 new active users and 200 signups equals an adoption rate of 20%.

**Is it useful?** We'd argue not. There are better, more specific measures for product health, like [new user activation](#new-user-activation).

### Customer retention rate

**What is it?** Customer retention rate is a classic metric that calculates the proportion of customers you retain from one period to the next. 

Early-stage companies might prefer to focus on customer retention, but revenue retention becomes more important once you've acquired a solid base of paying customers.

While you can calculate it as a monthly or annual metric, it's commonly visualized in a retention chart (aka retention matrix), which shows you how well customers retain over a longer periods of time.

> The retention rate formula is `((customers at end of period - new customers) / customers at start of period) x 100`. So, a business who began a month with 3,000 customers, acquired 150 new customers, and ended the month with 3,100 customers, would have retention rate of 98.3%.

Reviewer: ```suggestion
The retention rate formula is `((customers at end of period - new customers) / customers at start of period) x 100`. So, a business that begins a month with 3,000 customers, acquires 150 new customers, and ends the month with 3,100 customers, would have retention rate of 98.3%.
```

---
date: 2022-06-14
title: The most useful B2B SaaS product metrics (and how to measure them)
rootPage: /blog
sidebar: Blog
showTitle: true
hideAnchor: true
featuredImage: ../images/blog/give-back-friday.png
featuredImageType: full
categories: ["Guides", "PostHog Academy", "Product analytics"]
author: ["andy-vandervell"]
---

So, you're building a B2B SaaS product.

Everyone knows you've got to measure stuff to succeed. This is not news. 

What, why, and how? These are the questions that matter.

In this guide we will:

1. Reveal what makes a useful metric for a B2B product
2. Explain some of the common metrics used by B2B SaaS companies
3. Help you choose the right metrics for your product

> This guide is about _product_ metrics, rather than revenue or wider business metrics. Naturally, they're important too.

Author: ```suggestion
Please note: this guide is about _product_ metrics, rather than revenue or wider business metrics. Naturally, they're important too.
```

> Oh and, spoiler alert, we build an open-source analytics platform called PostHog. It's basically Mixpanel + Hotjar + Optimizely + LaunchDarkly in one tool you can host yourself. 

Reviewer 2: I reckon this should come later as it's a smidge too sales-y - you haven't justified to me yet why I should use a tool like PostHog in the article

Author: Would be interested in hearing alternative views on this. I put it here mainly in the interests of transparency, and tried to keep the language informal so it didn't feel too jarring. 

Alternatively, I could put it in the 'Recommended metrics' section as a footnote to that.

Author: They aren't, but we do replace them right? Sessions recording and feature flags.

Reviewer: Yeah, we do. Just think it makes it less sales-y if we stick to what's strictly relevant.

- [Choosing a North Star metric](#choosing-a-north-star-metric)
- [Common B2B product metrics](#common-b2b-saas-product-metrics)
- [Recommended B2B product metrics](#recommended-b2b-product-metrics)
- [Further reading](#further-reading)

> This article is part of our [PostHog Academy series](/blog/categories/posthog-academy) where we explain the fundamentals of product analytics.

## What makes a good B2B product metric?

Good metrics are understandable, comparative, specific and actionable.

Here's what those mean in a little more detail:

1. **Understandable:** Can you easily explain a metric to anyone in the business? If you can't, it's unlikely people will want to use or refer to it.

2. **Comparative:** Can you benchmark a metric against your industry, or compare internally using cohorts?

3. **Specific:** Some broad trends are useful, but we recommend metrics that measure specific actions. Don't measure how long users use your product, measure how long it takes for them to get value from it.

4. **Actionable:** Can you easily identify actions to take when you see a drop in a metric? If yes, you're onto a winner. Specific metrics are the most actionable metrics, too.

> ## Choosing a North Star metric

Reviewer 2: I wonder if the North Star section should be cut or maybe form part of 'further reading' under 'common B2B SaaS metrics'? It's a bit weird to have a guide and then say 'oh you now need to read another guide' and the section as a whole doesn't add anything.

Author: Hmm, yeah, this was orginally going to be a boxout. Agree it rather disrupts the flow. Might revert back to a boxout, or just remove entirely.

Every business can benefit from a single, North Star metric – ideally one several teams can impact.  Good North Star metrics provide an effective single focus for a company's success, and are incluenced by other more specific metrics.

There are numerous types of North Star metrics, such as revenue, consumption, engagement, growth efficiency, customer growth, and user experience. 

Which is best for you will depend on your product and the stage of your company, but we think a user experience metric is a good idea for most B2B SaaS products.

Read our guide to [finding your North Star metric](/blog/north-star-metrics) for more help with this.

> ## Common B2B SaaS product metrics

Reviewer 2: This whole section is the good stuff
